code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -138,6 +138,9 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 	radio = new /obj/item/radio/borg(src)
 	common_radio = radio
 
+	if(!cell) // Make sure a new cell gets created *before* executing initialize_components(). The cell component needs an existing cell for it to get set up properly
+		cell = new default_cell_type(src)
+
```

Comment:
Had to move this because of the godawful `init` proc you see below. This didn't cause any load order issues when I tested it.